<!-- ROSPlan turtlebot demo; requires turtlebot_gazebo -->
<launch>

	<!-- interface window -->
	<node name="explainable_planning_window" pkg="explainable_planning" type="main_window" respawn="false" output="screen">
	</node>

	<!-- arguments -->
	<arg name="domain"	default="$(find rosplan_demos)/common/domain_rover_time.pddl" />
	
	<!-- scene database (MongoDB) -->
	<include file="$(find mongodb_store)/launch/mongodb_store.launch">
		<arg name="db_path" value="$(find rosplan_knowledge_base)/common/mongoDB/"/>
	</include>

	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
		<param name="domain_path" value="$(arg domain)" />
		<param name="problem_path" value="$(find rosplan_demos)/common/pfile03_rover_time.pddl" />
	    <param name="database_path" value="$(find rosplan_knowledge_base)/common/mongoDB/" />
		<!-- conditional planning flags -->
		<param name="use_unknowns" value="False" />
	</node>

	<!-- problem generation -->
	<node name="rosplan_problem_interface" pkg="rosplan_planning_system" type="problemInterface" respawn="false" output="screen">
		<param name="domain_path" value="$(arg domain)" />
		<param name="problem_path" value="$(find rosplan_planning_system)/common/problem.pddl" />
		<param name="problem_topic" value="problem_instance" />
	</node>

	<!-- planning system -->
	<node name="rosplan_planner_interface" pkg="rosplan_planning_system" type="popf_planner_interface" respawn="false" output="screen">

		<!-- general options -->
		<param name="use_problem_topic" value="true" />

		<!-- ros topics -->
		<param name="problem_topic" value="/rosplan_problem_interface/problem_instance" />
		<param name="planner_topic" value="planner_output" />

		<!-- directory for files -->
		<param name="domain_path" value="$(arg domain)" />
		<param name="problem_path" value="$(find rosplan_planning_system)/common/problem.pddl" />
		<param name="data_path" value="$(find rosplan_planning_system)/common/" />

		<!-- to run the planner -->
		<param name="planner_command" value="$(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
	</node>

	<!-- plan parsing -->
	<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
		<param name="planner_topic" value="/rosplan_planner_interface/planner_output" />
		<param name="plan_topic" value="complete_plan" />
	</node>

	<!-- plan explanation -->
	<node name="rosplan_plan_explainer" pkg="explainable_planning" type="rpexplainer" respawn="false" output="screen">
		<param name="plan_topic" value="/rosplan_parsing_interface/complete_plan" />
		<param name="action_dispatch_topic" value="action_dispatch" />
		<param name="action_feedback_topic" value="action_feedback" />
	</node>

	<!-- sim actions -->
	<node name="rosplan_interface_navigate" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="navigate" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_recharge" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="recharge" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_sample_soil" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="sample_soil" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_sample_rock" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="sample_rock" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_drop" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="drop" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_calibrate" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="calibrate" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_take_image" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="take_image" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_communicate_soil_data" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="communicate_soil_data" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_communicate_rock_data" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="communicate_rock_data" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
	<node name="rosplan_interface_communicate_image_data" pkg="rosplan_planning_system" type="simulatedAction" respawn="false" output="screen">
		<param name="pddl_action_name" value="communicate_image_data" />
		<param name="action_dispatch_topic" value="/rosplan_plan_explainer/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_explainer/action_feedback" />
	</node>
</launch>

